lib/src/native/realm_core.dart

```diff
@@ -1405,6 +1481,16 @@ extension on Object {
   }
 }
 
+extension on List<AuthProviderType> {
+  AuthProviderType fromIndex(int index) {
+    if (!AuthProviderType.values.any((value) => value.index == index)) {
```

Comment:
Why is this an extension if it never accesses the object it extends? Should we use `this` instead of `AuthProviderType.values`?

Reply:
yes that's an extraordinary one. I wanted to extend `List<AuthProviderType>` but still enumerate all enum values. Otherwise the exception will not make sense. But I still wanted this method to be available on `AuthProviderType.values`.  
I guess I can change it to use the `this` but than the exception will mean that `the index was not found in this list only`. 
The current implementation means that every `List<AuthProviderType>` has a method that can convert an `int` to an `AuthProviderType`. To me it's reasonable to have extension methods which do not access `this`.

If you feel strongly about it I guess I could rework this to use `this` instead.

Reply:
No, I don't feel strongly about it, just found it a little awkward as I was reading it.